loader/loader.go

```diff
@@ -479,18 +482,37 @@ func (l *Loader) protoEnum(tspec *ast.TypeSpec) (*desc.EnumDescriptorProto, erro
 	return enum, nil
 }
 
-func (l *Loader) protoType(expr ast.Expr, pkg *types.Package) (desc.FieldDescriptorProto_Type, string) {
+func (l *Loader) protoType(expr ast.Expr, pkg *types.Package) (desc.FieldDescriptorProto_Type, desc.FieldDescriptorProto_Label, string) {
 	if pkg == nil {
 		pkg = l.tpkg
 	}
 	switch x := expr.(type) {
 	case *ast.Ident:
 		switch x.Name {
 		case "string":
-			return desc.FieldDescriptorProto_TYPE_STRING, x.Name
+			return desc.FieldDescriptorProto_TYPE_STRING, 0, x.Name
+		case "int", "int32":
+			return desc.FieldDescriptorProto_TYPE_INT32, 0, x.Name
+		case "bool":
+			return desc.FieldDescriptorProto_TYPE_BOOL, 0, x.Name
 		default:
 			fullName := "." + pkg.Path() + "." + x.Name
-			return desc.FieldDescriptorProto_TYPE_ENUM, fullName
+			// Check if the identifier is an already known 'message' or 'enum.
+			// NOTE: This checks the proto descriptor for messages and enums
+			// that have been previously added, this currently works because
+			// only top-level defined 'types' are expected.
+			// TODO: There is likely a much more robust way to handle this.
+			for _, et := range l.pfile.EnumType {
+				if *et.Name == x.Name {
+					return desc.FieldDescriptorProto_TYPE_ENUM, 0, fullName
+				}
+			}
+			for _, mt := range l.pfile.MessageType {
+				if *mt.Name == x.Name {
+					return desc.FieldDescriptorProto_TYPE_MESSAGE, 0, fullName
+				}
+			}
```

Comment:
Agreed that this can be cleaned up later. Fine to leave as it is for now with TODOs.